### ISSUE TITLE
Fatal error in version 8.5

### ISSUE THREAD

**Issue by superbmario** [woothemes/woocommerce#43406]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

```
Deprecated: Non-static method Automattic\WooCommerce\Internal\Features\FeaturesController::change_feature_enable() should not be called statically in /wp-content/plugins/woocommerce/src/Internal/Admin/Marketplace.php on line 23

Fatal error: Uncaught Error: Using $this when not in object context in /wp-content/plugins/woocommerce/src/Internal/Features/FeaturesController.php:340 Stack trace: #0 /wp-content/plugins/woocommerce/src/Internal/Admin/Marketplace.php(23): Automattic\WooCommerce\Internal\Features\FeaturesController::change_feature_enable('marketplace', true) #1 [internal function]: Automattic\WooCommerce\Internal\Admin\Marketplace->init() #2 /wp-content/plugins/woocommerce/src/Internal/DependencyManagement/Definition.php(59): call_user_func_array(Array, Array) #3 /wp-content/plugins/woocommerce/src/Internal/DependencyManagement/Definition.php(29): Automattic\WooCommerce\Internal\DependencyManagement\Definition->invokeInit(Object(Automattic\WooCommerce\Internal\Admin\Marketplace)) #4 /wp-content/plugins/woocommerce/lib/packages/Leagu in /wp-content/plugins/woocommerce/src/Internal/Features/FeaturesController.php on line 340

Deprecated: Non-static method Automattic\WooCommerce\Internal\Features\FeaturesController::change_feature_enable() should not be called statically in /wp-content/plugins/woocommerce/src/Internal/Admin/Marketplace.php on line 23

Fatal error: Uncaught Error: Using $this when not in object context in /wp-content/plugins/woocommerce/src/Internal/Features/FeaturesController.php:340 Stack trace: #0 /wp-content/plugins/woocommerce/src/Internal/Admin/Marketplace.php(23): Automattic\WooCommerce\Internal\Features\FeaturesController::change_feature_enable('marketplace', true) #1 [internal function]: Automattic\WooCommerce\Internal\Admin\Marketplace->init() #2 /wp-content/plugins/woocommerce/src/Internal/DependencyManagement/Definition.php(59): call_user_func_array(Array, Array) #3 /wp-content/plugins/woocommerce/src/Internal/DependencyManagement/Definition.php(29): Automattic\WooCommerce\Internal\DependencyManagement\Definition->invokeInit(Object(Automattic\WooCommerce\Internal\Admin\Marketplace)) #4 /wp-content/plugins/woocommerce/lib/packages/Leagu in /wp-content/plugins/woocommerce/src/Internal/Features/FeaturesController.php on line 340

```

### Expected behavior

N/A

### Actual behavior

White screen of death

### Steps to reproduce

not sure

### WordPress Environment

Can't get

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by BusinessBloomer** [comment]
Same: "NOTICE: PHP message: PHP Fatal error:  Uncaught Error: Using [$this](https://twitter.com/search?q=%24this&src=cashtag_click) when not in object context in /var/convesio/wordpress/wp-content/plugins/woocommerce/src/Internal/Features/FeaturesController.php:340"

For a quick downgrade to 8.4.0, I posted a quick tutorial on Twitter [here](https://twitter.com/rmelogli/status/1744683843485913534)

**Comment by superbmario** [comment]
Sad, I thought the days of fatal errors pushed to WC were behind us :(

To confirm, downgrading to 8.4.0 fixes it.

**Comment by BusinessBloomer** [comment]
Yes I just downgraded to 8.4.0. and site is back up

**Comment by BusinessBloomer** [comment]
Disabling automatic updates for now...

**Comment by Karthik-Naik** [comment]
WooCommerce -> Settings -> Advanced -> Features -> Marketplace

If this marketplace option is disabled in  8.4.0 or lower version of WooCommerce, and the plugin is updated then getting a fatal error.

If I enable this option and then update the plugin to 8.5, everything works fine.

**Comment by bgrgicak** [comment]
https://github.com/woocommerce/woocommerce/pull/43176 should fix this.

**Comment by Dge-06** [comment]
same, forced to downgrade to version 8.4 ... whait and see

**Comment by thomasjvu** [comment]
7558906-zen

**Comment by rwkyyy** [comment]
can confirm, similar to this

**Comment by rwkyyy** [comment]
> Disabling automatic updates for now...

automatic updates are never a good idea.

**Comment by Ginofelino** [comment]
I also have the same problem I went back to version 8.40 but I still have the same problem. I can't solve the problem the front end is offline.

**Comment by Babylon1999** [comment]
> WooCommerce -> Settings -> Advanced -> Features -> Marketplace
If this marketplace option is disabled in 8.4.0 or lower version of WooCommerce, and the plugin is updated then getting a fatal error.


If you've already upgraded while the option is disabled, you can enable it via the following WP-CLI command:

` wp --skip-plugins option update woocommerce_feature_marketplace_enabled yes`


Good catch btw @Karthik-Naik!

**Comment by pau1phi11ips** [comment]
> automatic updates are never a good idea.

Missing import security patches is generally not a good idea either 😉
Kinda expect something as big as WooCommerce to have decent testing.

**Comment by beamkiller** [comment]
Same here.

After updating to 8.5

`PHP Fatal error:  Uncaught Error: Non-static method Automattic\WooCommerce\Internal\Features\FeaturesController::change_feature_enable() cannot be called statically in /xxxx/wp-content/plugins/woocommerce/src/Internal/Admin/Marketplace.php:23
Stack trace:
#0 [internal function]: Automattic\WooCommerce\Internal\Admin\Marketplace->init()
#1 /xxxx/wp-content/plugins/woocommerce/src/Internal/DependencyManagement/Definition.php(59): call_user_func_array()
#2 /xxxx/wp-content/plugins/woocommerce/src/Internal/DependencyManagement/Definition.php(29): Automattic\WooCommerce\Internal\DependencyManagement\Definition->invokeInit()
#3 /xxxx/wp-content/plugins/woocommerce/lib/packages/League/Container/Definition/Definition.php(212): Automattic\WooCommerce\Internal\DependencyManagement\Definition->resolveClass()
#4 /xxxx/wp-content/plugins/woocommerce/lib/packages/League/Container/Definition/DefinitionAggregate.php(94): Automattic\WooCommerce\Vendor\League\Container\Definition\Definition->resolve()
#5 /xxxx/wp-content/plugins/woocommerce/lib/packages/League/Container/Container.php(157): Automattic\WooCommerce\Vendor\League\Container\Definition\DefinitionAggregate->resolve()
#6 /xxxx/wp-content/plugins/woocommerce/src/Internal/DependencyManagement/ExtendedContainer.php(158): Automattic\WooCommerce\Vendor\League\Container\Container->get()
#7 /xxxx/wp-content/plugins/woocommerce/lib/packages/League/Container/Container.php(178): Automattic\WooCommerce\Internal\DependencyManagement\ExtendedContainer->get()
#8 /xxxx/wp-content/plugins/woocommerce/src/Internal/DependencyManagement/ExtendedContainer.php(158): Automattic\WooCommerce\Vendor\League\Container\Container->get()
#9 /xxxx/wp-content/plugins/woocommerce/src/Container.php(116): Automattic\WooCommerce\Internal\DependencyManagement\ExtendedContainer->get()
#10 /xxxx/wp-content/plugins/woocommerce/includes/class-woocommerce.php(267): Automattic\WooCommerce\Container->get()
#11 /xxxx/wp-content/plugins/woocommerce/includes/class-woocommerce.php(189): WooCommerce->init_hooks()
#12 /xxxx/wp-content/plugins/woocommerce/includes/class-woocommerce.php(147): WooCommerce->__construct()
#13 /xxxx/wp-content/plugins/woocommerce/woocommerce.php(53): WooCommerce::instance()
#14 /xxxx/wp-content/plugins/woocommerce/woocommerce.php(68): WC()
#15 /xxxx/wp-settings.php(462): include_once('...')
#16 /xxxx/wp-config.php(127): require_once('...')
#17 /xxxx/wp-load.php(50): require_once('...')
#18 /xxxx/wp-admin/admin.php(34): require_once('...')
#19 /xxxx/wp-admin/index.php(10): require_once('...')
#20 {main}
  thrown in /xxxx/wp-content/plugins/woocommerce/src/Internal/Admin/Marketplace.php on line 23`
  
  This has helped:
  ` wp --skip-plugins option update woocommerce_feature_marketplace_enabled yes`

**Comment by shameemreza** [comment]
Another report here: 7558224-zen
Thread link: https://wordpress.org/support/topic/critical-error-solution-back-to-8-4

**Comment by rwkyyy** [comment]
> > WooCommerce -> Settings -> Advanced -> Features -> Marketplace
> > If this marketplace option is disabled in 8.4.0 or lower version of WooCommerce, and the plugin is updated then getting a fatal error.
> 
> If you've already upgraded while the option is disabled, you can enable it via the following WP-CLI command:
> 
> ` wp --skip-plugins option update woocommerce_feature_marketplace_enabled yes`
> 
> Good catch btw @Karthik-Naik!

might not work, it will crash the CLI response in some cases.

**Comment by rwkyyy** [comment]
> > automatic updates are never a good idea.
> 
> Missing import security patches is generally not a good idea either 😉 Kinda expect something as big as WooCommerce to have decent testing.

nothing about this release is a important security patch; I always give a few days and use tools that monitor the security of the websites; better this than have production broken by automatic updates.

**Comment by thisissandip** [comment]
7557974-zen

**Comment by Dge-06** [comment]
now it works, but my images and texts no longer appear. I use wpbakery and I think there is a conflict between these 2 plugins.

**Comment by thisissandip** [comment]
7559440-zen

**Comment by rossviviano** [comment]
7543370 - zen

**Comment by cardehngeh** [comment]
7557411 - zen

**Comment by cardehngeh** [comment]
7557409 - zen

**Comment by shameemreza** [comment]
Another report here: 7558189-zen
Thread link: https://wordpress.org/support/topic/update-causes-fatal-php-error/

Shared the WP-CLI workaround.

**Comment by wcstaley** [comment]
Wanted to note that this issue has been addressed and is fixed by #43428.  There is a mention of it in our [8.5 release notes](https://developer.woo.com/2024/01/09/woocommerce-8-5-0-released/) highlighting the known issue and the workaround available.  I am closing this issue for now given the above.
